addOns/domxss/src/main/java/org/zaproxy/zap/extension/domxss/DomXssScanRule.java

```diff
@@ -430,30 +455,54 @@ private void getHelper(WebDriverWrapper wrapper, String url, int retry) {
         return new ArrayList<>();
     }
 
+    private String getAlertDialogText() {
+        try {
+            org.openqa.selenium.Alert alertDialog = driver.getDriver().switchTo().alert();
+            if (alertDialog != null) {
+                return alertDialog.getText();
+            } else {
+                throw new Throwable();
+            }
+        } catch (Throwable e) {
+            // No alert
+            return "";
+        }
+    }
+
     private void scanHelper(String attackVector, String url) throws DomAlertException {
         if (this.isStop()) {
             return;
         }
 
+        String dialogText = "";
         try {
             getHelper(driver, url);
+            dialogText = getAlertDialogText();
         } catch (UnhandledAlertException uae) {
-            Stats.incCounter("domxss.vulns.get1");
-            throw new DomAlertException(url, attackVector);
+            if (dialogText.equals(UNLIKELY_STR)) {
```

Comment:
Should it call `getAlertDialogText()` here?

Reply:
No it has to be extracted before the exception is caught.

Reply:
The other cases should be updated accordingly then.

Reply:
I believe everywhere UnhandledAlertException is caught in scanHelper the value is extracted first.

Reply:
https://github.com/zaproxy/zap-extensions/pull/3108/files#diff-1a5f9d367076973762f350a1b9d1d30a89754fc7c2a8ba708703d6e6fa591eb2R516 and following are calling the method again instead of using `dialogText`, though https://github.com/zaproxy/zap-extensions/pull/3108#discussion_r698304952 still applies (will have to debug...).

Reply:
Oops okay I see what you're saying.

Reply:
Based on the debug it's ok to call `getAlertDialogText();` here, the text is extracted from the alert.